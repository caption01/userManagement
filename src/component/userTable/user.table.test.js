import React, {Component} from 'react';
import UserTable from './user.table.component';
import { shallow } from 'enzyme'

describe('Testing UserTable component', () => {

  it('expect to render UserTable', () => {
    const wrapper = shallow(<UserTable />)
    expect(wrapper).toMatchSnapshot()
  })

  it('testing handleModalSelect to set selected modal state correctly', () => {

    const mockProps = {
      setEditUserObj: jest.fn(),
      setActiveModal: jest.fn(),
    }
    const mockData = {
      key: '10',
      first_name: 'john',
      last_name: 'mayer',
      address: 'USA'
    }
    const wrapper = shallow(<UserTable {...mockProps} />)

    wrapper.instance().handleModalSelect('modal_edit', mockData)
    expect(wrapper.state().selected_modal).toEqual("modal_edit")

    wrapper.instance().handleModalSelect('modal_new_user', mockData)
    expect(wrapper.state().selected_modal).toEqual("modal_new_user")

  })

})

