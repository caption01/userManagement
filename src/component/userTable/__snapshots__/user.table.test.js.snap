// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserTable component expect to render UserTable 1`] = `
<div
  className="user-table-container"
>
  <div
    className="user-table-add-button"
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      id="create-new-user-btn"
      loading={false}
      onClick={[Function]}
      type="primary"
    >
      Add new User
    </Button>
  </div>
  <div
    className="user-table"
  >
    <withStore(Table)
      columns={
        Array [
          Object {
            "dataIndex": "first_name",
            "key": "name",
            "title": "First Name",
          },
          Object {
            "dataIndex": "last_name",
            "key": "age",
            "title": "Last Name",
          },
          Object {
            "dataIndex": "address",
            "key": "address",
            "title": "Address",
          },
          Object {
            "key": "action",
            "render": [Function],
            "title": "Action",
          },
        ]
      }
    />
  </div>
</div>
`;
